File: 011-ReplaceCharacter Function/ReplaceCharacterFunction.cpp
#include<iostream>
#include<string>
using namespace std;

string ReplaceCharacter(string str, char ch1, char ch2){
    if(str.empty()){
        return "";
    }

    if(ch1 == ch2){
        return str;
    }
    bool found_ch1 = false;
    bool found_ch2 = false;

    for(int i = 0; i < str.length(); i++){
        if(str[i] == ch1){
            found_ch1 = true;
        }
        if(str[i] == ch2){
            found_ch2 = true;
        }   
    }
    if(!found_ch1 && !found_ch2){
        return str;
    }

    for(int  i = 0; i < str.length(); i++){
        if(str[i] == ch1){
            str[i] = '#';
        }else if(str[i] == ch2){
            str[i] = ch1;
        }
    }

    for (int i = 0; i < str.length(); i++) {
        if (str[i] == '#') {
            str[i] = ch2;
        }
    }

    return str;
}
int main() {
    string str = "apples";
    char ch1 = 'a';
    char ch2 = 'p';

    string result = ReplaceCharacter(str, ch1, ch2);

    cout << "Original String: " << str << endl;
    cout << "Characters to swap: '" << ch1 << "' and '" << ch2 << "'" << endl;
    cout << "Modified String: " << result << endl;

    return 0;
}

File: 011-ReplaceCharacter Function/ReplaceCharacterFunction.py
def replace_character(s: str, ch1: str, ch2: str) -> str:
    # Return None if the input is None
    if s is None:
        return None

    # If both characters are the same, return the original string
    if ch1 == ch2:
        return s

    # Check if ch1 or ch2 exist in the string
    found_ch1 = False
    found_ch2 = False

    for i in range(len(s)):
        if s[i] == ch1:
            found_ch1 = True
        elif s[i] == ch2:
            found_ch2 = True

    # If neither ch1 nor ch2 are present, return the original string
    if not found_ch1 and not found_ch2:
        return s

    # First pass: Replace ch1 with '#' (temporary), ch2 with ch1
    temp = []
    for i in range(len(s)):
        if s[i] == ch1:
            temp.append('#')  # temporary placeholder
        elif s[i] == ch2:
            temp.append(ch1)
        else:
            temp.append(s[i])

    # Second pass: Replace '#' with ch2
    for i in range(len(temp)):
        if temp[i] == '#':
            temp[i] = ch2

    # Construct the final string
    result = ""
    for i in range(len(temp)):
        result += temp[i]

    return result

# Example usage
s = "apples"
ch1 = 'a'
ch2 = 'p'

result = replace_character(s, ch1, ch2)

print("Original String:", s)
print("Characters to swap:", ch1, "and", ch2)
print("Modified String:", result)
